snuba/admin/clickhouse/system_queries.py

```diff
@@ -200,19 +232,30 @@ def run_system_query_on_host_with_sql(
         if not can_sudo:
             raise UnauthorizedForSudo()
 
-    if is_query_select(system_query_sql):
-        validate_system_query(system_query_sql)
-    elif is_query_describe(system_query_sql):
-        pass
-    elif is_query_show(system_query_sql):
-        pass
-    elif sudo_mode and (
-        is_system_command(system_query_sql)
-        or is_query_alter(system_query_sql)
-        or is_query_optimize(system_query_sql)
+    def is_valid_query(
+        clickhouse_host: str,
+        clickhouse_port: int,
+        storage_name: str,
+        system_query_sql: str,
+        sudo_mode: bool,
+    ) -> bool:
+        if not sudo_mode and is_valid_system_query(
+            clickhouse_host, clickhouse_port, storage_name, system_query_sql
+        ):
+            return True
```

Comment:
Should we return specific error messages from validation functions. Raising InvalidCustomQuery with detailed error messages

Reply:
I think we should do that, but it's kind of hard to have detailed error messages when we don't know what kind of query the user is trying to write.

Reply:
I agree that this would be nice to have. Then if users enter an invalid query they could quickly understand why its invalid.